src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmRpoServiceImpl.java

```diff
@@ -20,6 +22,19 @@ public class ArmRpoServiceImpl implements ArmRpoService {
 
     public static final String ARM_RPO_EXECUTION_DETAIL_NOT_FOUND = "ArmRpoExecutionDetail not found";
     private final ArmRpoExecutionDetailRepository armRpoExecutionDetailRepository;
+    private final EntityManager entityManager;
+
+    @Override
+    @Transactional
+    public ArmRpoExecutionDetailEntity createArmRpoExecutionDetailEntity(UserAccountEntity userAccount) {
+        ArmRpoExecutionDetailEntity armRpoExecutionDetailEntity = new ArmRpoExecutionDetailEntity();
+
+        UserAccountEntity mergedUserAccountEntity = entityManager.merge(userAccount);
```

Comment:
Is this needed?

Reply:
I think it's needed, but there were some challenges in this area, so I'd be interested in your perspective here.

The `userIdentity.getUserAccount()` call that obtains the `UserAccountEntity` takes place within `TriggerArmRpoSearchServiceImpl.triggerArmRpoSearch()`, as we must pass the entity to a series of method calls.

There is no active transaction within `triggerArmRpoSearch()`, so upon completion of `userIdentity.getUserAccount()`, our entity is no longer attached to an entity manager. A new transaction is then started here in `createArmRpoExecutionDetailEntity()`, and we merge the `userAccount` to the entity manager prior to saving the `ArmRpoExecutionDetailEntity` to avoid detached entity exceptions.

I deliberately avoided making the higher level method `triggerArmRpoSearch()` `@Transactional` on the basis that the underlying method calls (`getRecordManagementMatter()`, `getIndexesByMatterId()` etc) explicitly handle exceptions by setting the state and a failure status on the corresponding `ArmRpoExecutionDetailEntity`. And so, if the `triggerArmRpoSearch()` is made transactional, in the event of an exception bubbling up we will get the unwanted side-effect of the transaction rolling back the DB state, undoing those statuses.

There were a couple of other approaches that I thought about:
* Explictly create a more limited transaction using `TransactionTemplate`, wrapping the `userIdentity.getUserAccount()` and `armRpoService.createArmRpoExecutionDetailEntity()` calls. This would have added a little more code complexity than the merge aprpoach, but I think is feasible.
* Make `triggerArmRpoSearch()` transactional, and wrap all method calls in a try/catch, and catch `Exception` to avoid any exceptions that bubble up from triggering a rollback. For some reason I don't understand this approach caused constraint violation issues whenever an exception was thrown, and so state wasn't properly recorded in the DB.
* Make `triggerArmRpoSearch()` transactional, with `@Transactional(noRollbackFor = Exception.class)` but this doesn't appear to include all subsclasses of exception.

So on balance, I went with the merge approach as the simplest solution. But to be honest I'm a little unsure why other automated tasks don't need to take this approach, as they aren't run inside a transaction either. So I might be missing something...